### ISSUE TITLE
Is the constructor of AbstractDecompilerAction intentionally only accessible from the package

### ISSUE THREAD

**Issue by utnubu1** [NationalSecurityAgency/ghidra#1655]
I am trying to add a custom popup menu entry to the decompiler view. I used https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/decompile/actions/SpecifyCPrototypeAction.java as reference. 

Trying to implement my Plugin similar to the reference code i found that the constructor of the AbstractDecompilerAction class is only accessible from within the ghidra.app.plugin.core.decompile.actions package. 
Is this intentionally ? And if so what is the appropriate way to add own menu entrys to the decompiler view ?

**Comment by dragonmacher** [comment]
The constructor does not need to be package, it was just refactored that way.

To get your actions working, you can simply copy whatever code is inside of that class and mimic it in your action.

> And if so what is the appropriate way to add own menu entrys to the decompiler view ?

Ghidra actions work by using **ActionContext**.   You can look at existing actions for examples.  In your plugin, register your actions in the tool via `PluginTool.addAction()`.  Then, your action enablement is predicated upon the current context.  You can see how the existing `Decompiler` actions use context to determine enablement.  As far as the popup goes, also see these actions to get a sense of how they work.

**Comment by buffaloverflow** [comment]
Ok, I finally figured out to use `ProgramLocationContextAction` as the Action type since `DecompilerAction `/ `DecompilerContextAction` doesn't exist / isn't public.

**Comment by fmagin** [comment]
@dragonmacher I'm writing various plugins which tend to contain various decompiler actions, so it would be fairly helpful for me if the constructor for this action was public. Copying the code to each plugin is annoying boilerplate. One workaround is just moving my actions into the `package ghidra.app.plugin.core.decompile.actions` (but still in my plugin), but this does not seem clean either. 

The `AbstractDecompilerAction`  is less useful with the new `ActionBuilder` (because `withContext` already covers much of the benefits), but the additional helpers are quite nice, and the generic types in `ActionBuilder` don't seem to work with Kotlin for unclear reasons, so I do need the `AbstractDecompilerAction` still.

Are their concrete concerns preventing this from being public?

**Comment by fmagin** [comment]
> I'll make this change.

Thank you!

**Comment by dragonmacher** [comment]
In Java the default scope is 'package'.  Protected allows for extending classes to call the parent.

**Comment by fmagin** [comment]
> In Java the default scope is 'package'. Protected allows for extending classes to call the parent.
Yeah, I deleted this comment  already because I remembered that there was some subtle difference about this, should have re-checked before commenting first
